libs/facilities/backend/timeseries/src/lib/services/timeseries.service.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { faker } from '@faker-js/faker';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Prisma } from '@prisma/client';
+import { PrismaService } from 'common-backend-prisma';
+import { lastValueFrom } from 'rxjs';
+
+import { IGetTimeSeriesParams, IGetTimeseriesQuery } from '../interfaces/request.interface';
+import { XdTimeseriesService } from './timeseries.service';
+
+describe('TimeseriesService', () => {
+	let service: XdTimeseriesService;
+	let prisma: PrismaService = new PrismaService();
+
+	beforeEach(async () => {
+		const prismaServiceMock = {
+			onModuleInit: jest.fn(),
+
+			selectKeysFromJSON: prisma.selectKeysFromJSON,
+			isTTimeSeriesData: prisma.isTTimeSeriesData,
+			timeSeriesDataItem: {
+				findMany: jest.fn().mockImplementation(() => [
+					{
+						time: new Date(),
+						data: JSON.stringify({ test: 'test', test2: 'test2' }),
+					},
+				]),
+			},
+		};
+
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [
+				XdTimeseriesService,
+				{
+					provide: PrismaService,
+					useValue: prismaServiceMock,
+				},
+			],
+		}).compile();
+
+		service = module.get<XdTimeseriesService>(XdTimeseriesService);
+		prisma = module.get<PrismaService>(PrismaService);
+	});
+
+	afterAll(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should be defined', () => {
+		expect(service).toBeDefined();
+	});
+
+	it('should return the correct time series data', async () => {
```

Comment:
Maybe wrap all tests inside describe 'getTimeSeriesFromDB'

If other methods should be added over time theres a better overview

Reply:
True